### PR TITLE
Optional prewarm logic to speed up initial queries

### PR DESCRIPTION
**Pull request by TimMcCauley**
This PR adds logic to enable prewarming in postgres ([link2](https://www.cybertec-postgresql.com/en/prewarming-postgresql-i-o-caches/), [link3](https://blog.dbi-services.com/pre-warming-the-buffer-cache-in-postgresql/)) #95 .

Tested on Ubuntu 18.04 with PostgreSQL 11  and PostGIS 2.5.

Note 1: [auto-prewarming will only work from postgres11 onwards.](https://www.enterprisedb.com/blog/autoprewarm-new-functionality-pgprewarm)

Note 2: I can't get it working in the same way inside a Docker container (see https://github.com/kartoza/docker-postgis/issues/229). Let's keep this PR open until we know more about this.

Using `415430` POIs from

```
178M	minnesota-latest.osm.pbf
71M	mississippi-latest.osm.pbf
62M	montana-latest.osm.pbf
84M	new-mexico-latest.osm.pbf
64M	puerto-rico-latest.osm.pbf
```

As a test with a query such as 

```
\timing

SELECT count(*) FROM ops_pois p WHERE ST_Within(p.geom::geometry,ST_GeomFromGeoJSON('{"type":"Polygon","coordinates":[[[-106.5363109731028,35.151747622563988],[-106.532802489259481,35.118222110283412],[-106.556192381548257,35.119391604897856],[-106.590887388443278,35.133425540271119],[-106.56009069692972,35.159934084865064],[-106.552294066166795,35.145510317953651],[-106.543717772327582,35.139662844881457],[-106.5363109731028,35.151747622563988]]],"crs":{"type":"name","properties":{"name":"EPSG:4326"}}}'));
```

Will take about 350-400ms in the first initial query. Subsequent queries will be double the speed as the cache will be filled. Enabling prewarming will speed up the first initial query the same way. 

If you want to test, you can use this script to drop the caches and time the query.

```
#!/usr/bin/sudo bash

service postgresql stop
sync
echo 3 > /proc/sys/vm/drop_caches
service postgresql start
```

### CODE REVIEW

File: manage.py [hunk 1]
```diff
@@ -57,6 +58,17 @@ def import_data():
 
     logger.info('Starting to import OSM data from\n\t{}'.format("\n\t".join([str(p.resolve()) for p in osm_files])))
     parser.run_import(osm_files)
+
+    prewarm = app.config['POSTGRES_PREWARM']
+    if not is_testing() and prewarm == True:
+        tables = ['ops_pois', 'ops_categories', 'ops_tags']
```
Comment (TimMcCauley) [inline manage.py:64]: I tried adding the table names to the app config itself but because flasks app context is missing in `models.py` they can't be used there so I hardcoded them..

File: README.md [hunk 2]
```diff
@@ -64,13 +64,14 @@ Best is to set up a `.env` file with the settings that need changing (e.g. `POST
 
 ##### PostgreSQL
 
-The PostgreSQL specific settings are the same as for Kartoza's excellent [PostGIS image](https://github.com/kartoza/docker-postgis):
+The PostgreSQL specific settings required to configure your database connection:
```
Comment (nilsnolde) [inline README.md:67]: Why remove? I think that's important information for any `docker-compose` setup, as you only have to create one `.env` file.

Comment (nilsnolde) [reply README.md:67]: Maybe rephrase yes. Maybe you misunderstood, as it's not very clear: the env var names are the same as in kartoza's image.

**Comment by nilsnolde** [comment]
Thanks for this. We can merge now, as you're PRing into an active dev branch. Otherwise it's likely that everything changes.

**Comment by nilsnolde** [comment]
When I restart the postgres container, does it "need" a prewarm? If so, maybe it's better to have a separate `manage` command for prewarming? then there wouldn't be a need for the env var I guess.

**Comment by TimMcCauley** [comment]
No, it doesn't. It auto prewarms which is a separate process. Not sure yet reg Docker, waiting for Kartoza to reply.

File: manage.py (hunk 1 above)
Comment (TimMcCauley) [reply manage.py]: Yep, thanks, done

File: README.md (hunk 2 above)
Comment (TimMcCauley) [reply README.md:67]: Yes but no need to point that out? Above it's recommended to use Kartoza, that should be enough.

File: openpoiservice/utils/osm_reader.py [hunk 3]
```diff
@@ -86,18 +86,20 @@ def area(self, obj_area):
...
-            for poly in geom_multipolygon:
-                self.store_objects(
-                    osmid=osmid,
-                    tags=tags,
-                    location=poly.centroid.wkt,
-                    osm_type=osm_type,
-                    categories=cat_included
-                )
-
+            try:
```
[the remark this answers is not in the document]
Comment (TimMcCauley) [reply openpoiservice/utils/osm_reader.py:89]: It is because of inner polygon rings, see here: https://github.com/osmcode/libosmium/issues/225 
I will add this to the comment inline and add another issue.

[the remark this answers is not in the document]
Comment (TimMcCauley) [reply openpoiservice/utils/osm_reader.py:89]: Actually, I won't add another issue, inner rings aren't needed, so the try/catch can stay there.

[the remark this answers is not in the document]
Comment (TimMcCauley) [reply openpoiservice/utils/osm_reader.py:89]: Sorry wrong, we do want to handle polygons with inner rings but just use the outer. Will add an issue..

[the remark this answers is not in the document]
Comment (TimMcCauley) [reply openpoiservice/utils/osm_reader.py:89]: In imposm world we did it this way: https://github.com/GIScience/openpoiservice/blob/master/openpoiservice/server/db_import/parse_osm.py#L112

File: config_flask.py [hunk 4]
```diff
@@ -14,9 +15,10 @@ class BaseConfig(object):
     # Database
     POSTGRES_HOST = os.environ.get('POSTGRES_HOST', 'localhost')
     POSTGRES_DBNAME = os.environ.get('POSTGRES_DBNAME', 'gis')
-    POSTGRES_PORT = int(os.environ.get('POSTGRES_PORT', '5432'))
-    POSTGRES_USER = os.environ.get('POSTGRES_USER', 'gis_admin')
-    POSTGRES_PASS = os.environ.get('POSTGRES_PASS', 'admin')
+    POSTGRES_PORT = int(os.environ.get('POSTGRES_PORT', '5433'))
+    POSTGRES_USER = os.environ.get('POSTGRES_USER', 'gisops')
+    POSTGRES_PASS = os.environ.get('POSTGRES_PASS', 'gisops')
```
Comment (nilsnolde) [inline config_flask.py:20]: Hm, just seeing this, since Travis failed. Can we revert this again? If you want local overrides, create a `.env` or put it in `.flaskenv`.

Comment (TimMcCauley) [reply config_flask.py]: Yep, done

**Comment by TimMcCauley** [comment]
Added in https://github.com/kartoza/docker-postgis/pull/231/files therefore merging here!
